### DISCUSSION TITLE
Track form for analytics

### DISCUSSION THREAD

**Discussion by thomaschaaf** [react-hook-form/react-hook-form#2334] [Q&A]
I would like to track when a field is changed and trigger an analytics function so that we can track where people are stopping inputting data. Is there a way to listen to all changes in a form and only get the changes?

Right now I am considering using watch() and comparing the current values to the value to the previous value.

**Comment by bluebill1049** [comment]
watch/useWatch (prefer useWatch) or attach onChange on each input manually.
